<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>876</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string> Mesh Spectrum Representation</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>VIBOT.jpg</normaloff>VIBOT.jpg</iconset>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Triangular</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>881</width>
      <height>521</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="RenderingTab">
     <attribute name="title">
      <string>Mesh Rendering</string>
     </attribute>
     <widget class="myGLWidget" name="RenderingView" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>501</width>
        <height>511</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::StrongFocus</enum>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_CompTime">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>360</y>
        <width>351</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>Computation time information</string>
      </property>
      <widget class="QTextEdit" name="progressEditor">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>331</width>
         <height>91</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QCheckBox" name="checkBox">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>40</y>
        <width>121</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Use default setting</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="CamgroupBox">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>60</y>
        <width>171</width>
        <height>131</height>
       </rect>
      </property>
      <property name="title">
       <string>Camera position</string>
      </property>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>X :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Y :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Z:</string>
       </property>
      </widget>
      <widget class="QSlider" name="zHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>111</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="pageStep">
        <number>10</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBelow</enum>
       </property>
      </widget>
      <widget class="QSlider" name="yHorizontalSlider_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>111</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="pageStep">
        <number>10</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBelow</enum>
       </property>
      </widget>
      <widget class="QSlider" name="xHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>111</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="pageStep">
        <number>10</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBelow</enum>
       </property>
      </widget>
     </widget>
     <widget class="QTextEdit" name="filePathEditor">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>610</x>
        <y>10</y>
        <width>251</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="IntensitylineEdit">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>340</y>
        <width>41</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>520</x>
        <y>340</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Light intensity : </string>
      </property>
     </widget>
     <widget class="QGroupBox" name="DiffgroupBox">
      <property name="geometry">
       <rect>
        <x>740</x>
        <y>190</y>
        <width>121</width>
        <height>141</height>
       </rect>
      </property>
      <property name="title">
       <string>Diffuse lighting</string>
      </property>
      <widget class="QLabel" name="label_15">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>R</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_18">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>G</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_19">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>B</string>
       </property>
      </widget>
      <widget class="QSlider" name="DiffRHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>20</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QSlider" name="DiffGHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>50</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QSlider" name="DiffBHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>80</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
       <property name="tickInterval">
        <number>10</number>
       </property>
      </widget>
      <widget class="QSlider" name="DiffAHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>110</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QLabel" name="label_20">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>A</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="SpecgroupBox">
      <property name="geometry">
       <rect>
        <x>630</x>
        <y>190</y>
        <width>111</width>
        <height>141</height>
       </rect>
      </property>
      <property name="title">
       <string>Specular lighting</string>
      </property>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>R</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>G</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>B</string>
       </property>
      </widget>
      <widget class="QSlider" name="SpecRHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>20</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QSlider" name="SpecGHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>50</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QSlider" name="SpecBHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>80</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="singleStep">
        <number>0</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
       <property name="tickInterval">
        <number>10</number>
       </property>
      </widget>
      <widget class="QSlider" name="SpecAHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>110</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>A</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="pushButton">
      <property name="geometry">
       <rect>
        <x>780</x>
        <y>340</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Render</string>
      </property>
     </widget>
     <widget class="QPushButton" name="searchButton">
      <property name="geometry">
       <rect>
        <x>520</x>
        <y>10</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Open file</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="AmgroupBox">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>190</y>
        <width>121</width>
        <height>141</height>
       </rect>
      </property>
      <property name="title">
       <string>Ambient lighting</string>
      </property>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>R</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>G</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>B</string>
       </property>
      </widget>
      <widget class="QSlider" name="AmRHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>20</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QSlider" name="AmGHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>50</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QSlider" name="AmBHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>80</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
       <property name="tickInterval">
        <number>10</number>
       </property>
      </widget>
      <widget class="QSlider" name="AmAHorizontalSlider">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>110</y>
         <width>71</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::NoTicks</enum>
       </property>
      </widget>
      <widget class="QLabel" name="label_17">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>21</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>A</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>60</y>
        <width>181</width>
        <height>131</height>
       </rect>
      </property>
      <property name="title">
       <string>Spectrum analysis</string>
      </property>
      <widget class="QRadioButton" name="GeodesicSpectrumradioButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>20</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Geodesic spectrum</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="UniformradioButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Uniform weight</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="CotangentradioButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>141</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Cotangent weight</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="MeanradioButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>131</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Mean value coordinate</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>100</y>
         <width>71</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Eigenvector :</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="eigenSpinBox">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>100</y>
         <width>42</width>
         <height>22</height>
        </rect>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="ManualTab">
     <attribute name="title">
      <string>Manual</string>
     </attribute>
     <widget class="QPlainTextEdit" name="ManualEditor">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>871</width>
        <height>491</height>
       </rect>
      </property>
      <property name="plainText">
       <string>How to  render 3D objects :
     
    1. Go to 'Mesh Rendring' Tab to choose a file 
              Click on 'Open File' button ,and choose a &quot;.wrl&quot; file.

     2. Choose the camera and lighting setting
             Adjust the camera and lighting setting by changing the parameter given. Checking the &quot;Use default setting&quot; checkbox will set the setting to be default.
            a. Default setting for camera  :
                      - X-Position : 0
                      - Y-Position : 0
                      - Z-Position : three times the distance from the center of the object
           b. Default setting for lighting
                     -  shining intensity is 100 
                     - ambient[R,G,B,a] = {0.1, 0.1, 0.1, 1.0}
                     - specular[R,G,B,a] = {1.0, 1.0, 1.0, 1.0}
                     - diffuse[R,G,B,a] = {0.4, 0.4, 0.4, 1.0}

      3. Choose rendering type
             1. Geodesic spectrum 
                    The distance is calculated by calculating the number of edges in a shortest path (also called a graph geodesic) connecting them. 
             2. Laplacian spectrum.
                    a. Uniform weight
                    b. Cotangent weight
                    c. Mean value weight
               Warning : Use small mesh with ( &lt;10000 vertices ). The computation to load the representation is quite heavy. 
              Please wait when choosing this representation as heavy computation is running on the background.
      
    4. Render
         Click the &quot;Render&quot; Button

    Detail computation time will be shown in 'Computation time box' after the rendering finished. 

   Keyboard Shortcut :
     - L :  draw the mesh in Polygon mode                            left_arrow     : move left
     - F :  draw the mesh in Normal mode                              right_arrow  : move right
     - -  : Zoom out                                                                 up_arrow     : move up
     - + : Zoom in                                                                   down_arrow : move down
              
******************************************************************************************************************************************</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="AboutTab">
     <attribute name="title">
      <string>About</string>
     </attribute>
     <widget class="QPlainTextEdit" name="AboutEditor">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>881</width>
        <height>491</height>
       </rect>
      </property>
      <property name="plainText">
       <string>******************************************************************************************************************************************
                                  		                                    Spectral Mesh Representation
                           		                                       -------------------
   			                       update             : 2014 - 01 - 14
    			                      copyright          : (C) 2013 by Pramita Winata
   			                      email                 : pramita.winata@gmail.com
 
   This application is done as part of Software Engineering course project.  The aim of this project is to show basic Laplace spectral representation and implement C++    programming skill.  It is developed under Qt IDE 5.3.0 using OpenGL and Eigen library. I will discuss the project implementation in two main sections; basic laplace representation and application development.  &quot;Laplacian Mesh Processing&quot; paper by Olga Sorkine is mainly used in the Laplace implementation. The development of the application is the extension of the existing project where improvement in the Graphical User Interface (GUI) is done and Laplacian representation is cooperated. 


 ******************************************************************************************************************************************
</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>myGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">myglwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
